src/lib/mongoBackend/dbFieldEncoding.cpp

```diff
@@ -41,8 +41,9 @@
 * Replace:
 *   . => =
 */
-std::string dbDotEncode(std::string s)
+std::string dbDotEncode(const std::string& _s)
 {
+  std::string s = _s; // replace cannot be used in const std::string&
   std::replace(s.begin(), s.end(), ESCAPE_1_DECODED, ESCAPE_1_ENCODED);
   return s;
 }
```

Comment:
Not sure if I get the point... using `const std::string& _s` is a way of ensuring that `s` cannot be modified back to the caller. That's is the reason of the copy done by `s = s_`.

I mean, I don't see the need of using a pointer when `const std:string&` suffices the case (as far as I understand).